components/sync15/src/request.rs

```diff
@@ -301,6 +321,8 @@ where
         // The + 1 is only relevant for the final record, which will have a trailing ']'.
         let item_len = item_end - item_start + 1;
 
+        // Catches errors if the item being put into the batch
```

Comment:
Same here. This code is actually handling the error where the record is too large to ever be updated.

The check on line 284 is the result of `payload_length`, which is supposed to provide an overestimate. This check is for the actual serialized length, which may be less than `payload_length`.

`// Check if the actual serialized length of the record is too long to ever be submitted with current server limits.`